### ISSUE TITLE
Profile password field in the wrong place

### ISSUE THREAD

**Issue by redmanmale** [uTox/uTox#821]
After click `Show password field` button:

![pic](http://puu.sh/viSt6/4420d61e70.png)

It's probably because of this [code here](https://github.com/uTox/uTox/blob/develop/src/ui.c#L209):
`CREATE_EDIT(profile_password, SCALE(10),  SCALE(88) + (SCALE(167) * panel_profile_password.disabled), SCALE(-10), SCALE(24));`

May I ask, what is the purpose of `SCALE(88) + (SCALE(167) * panel_profile_password.disabled)`?

If I change it just to `SCALE(255)`, like the other definitions, it looks good again.

**Env**
Ubuntu 16.04, uTox v.0.14.0 and latest develop (b07c290e1d2e42c37d39206215b6323ad4a025d0).
Only on Ubuntu, not on Windows.

P.S. I'm working on fixes for some other minor UI flaws and could fix this one too, if I get this right.

**Comment by GrayHatter** [comment]
have you tested a simple scale(255) with a password set?

**Comment by redmanmale** [comment]
There's a problem with it too.
I was going to check it on a clean system, because on my Ubuntu I coundn't set password at all.
I run uTox with logging and got warning message from toxcore that profile cound not be encrypted and saved in plain text. I could get exact message tomorrow and find where it comes from.

**Comment by GrayHatter** [comment]
I'll get some time and take a look

@endoffile78 maybe you can take a look too...

**Comment by endoffile78** [comment]
I just tested it too and changing it to SCALE(255) does fix it in the settings page but its not in the right place when you open uTox. I'll look into it more.

**Comment by redmanmale** [comment]
It seems that `panel_profile_password.disabled` flag isn't properly set after loading profile (with or without password).

Also maybe we should use different edit controls for password in login page and in settings.

**Comment by redmanmale** [comment]
Also on Windows with password profile.
Without password all is OK.

**Comment by GrayHatter** [comment]
@redmanmale does this fix it for you?

```diff
diff --git a/src/layout/settings.c b/src/layout/settings.c
index 47543984..a7e8e2b9 100644
--- a/src/layout/settings.c
+++ b/src/layout/settings.c
@@ -268,12 +268,29 @@ static void draw_profile_password(int UNUSED(x), int UNUSED(y), int UNUSED(w), i
 }
 
 PANEL
+boxfor_password_entry_login = {
+    .type = PANEL_NONE,
+    .x = 0, .y = 0,
+    .child = (PANEL*[]) {
+        (PANEL*)&edit_profile_password,
+        NULL
+    }
+},
+boxfor_password_entry_change = {
+    .type = PANEL_NONE,
+    .x = 0, .y = 180,
+    .child = (PANEL*[]) {
+        (PANEL*)&edit_profile_password,
+        NULL
+    }
+},
+
 panel_profile_password = {
     .type = PANEL_NONE,
     .disabled = 0,
     .drawfunc = draw_profile_password,
     .child = (PANEL*[]) {
-        (PANEL*)&edit_profile_password,
+        (PANEL*)&boxfor_password_entry_login,
         NULL
     }
 },
@@ -294,7 +311,7 @@ panel_profile_password_settings = {
     .disabled = 1,
     .drawfunc = draw_settings_text_password,
     .child = (PANEL*[]) {
-        (PANEL*)&edit_profile_password,
+        (PANEL*)&boxfor_password_entry_change,
         (PANEL*)&button_lock_uTox,
         NULL
     }
diff --git a/src/ui.c b/src/ui.c
index cf6cd797..185869d9 100644
--- a/src/ui.c
+++ b/src/ui.c
@@ -206,7 +206,7 @@ static void settings_ADV(void) {
     CREATE_BUTTON(change_nospam,  SCALE(10),  SCALE(285), BM_SBUTTON_WIDTH, BM_SBUTTON_HEIGHT);
     CREATE_BUTTON(revert_nospam,  SCALE(200), SCALE(285), BM_SBUTTON_WIDTH, BM_SBUTTON_HEIGHT);
 
-    CREATE_EDIT(profile_password, SCALE(10),  SCALE(88) + (SCALE(167) * panel_profile_password.disabled), SCALE(-10), SCALE(24));
+    CREATE_EDIT(profile_password, SCALE(10),  SCALE(88), SCALE(-10), SCALE(24));
     CREATE_BUTTON(lock_uTox,      SCALE(10),  SCALE(285), BM_SBUTTON_WIDTH, BM_SBUTTON_HEIGHT);
 }
```

[password.txt](https://github.com/uTox/uTox/files/922930/password.txt)

**Comment by GrayHatter** [comment]
password.txt should be saved as password.diff

**Comment by redmanmale** [comment]
@GrayHatter 
Yes. Your patch fixes this issue.

**Comment by GrayHatter** [comment]
@robinlinden check windows and then ping me? I'll commit it and open a pr with this. Or you could just push after appling the patch

**Comment by redmanmale** [comment]
I checked it on Windows too.

**Comment by robinlinden** [comment]
Fixed in #835
